src/tests/EventHistogramWriterTests.cpp

```diff
@@ -300,10 +299,9 @@ createTestMessage(size_t HistogramID, size_t PacketID,
 
   flatbuffers::Offset<void> DataValue;
   {
-    size_t TotalElements = 1;
-    for (auto x : ThisLengths) {
-      TotalElements *= x;
-    }
+    size_t TotalElements = std::accumulate(
+        ThisLengths.begin(), ThisLengths.end(), 1, std::multiplies<>());
```

Comment:
`std::accumulate` is a pain. I think `1` is treated as an `int` here, then the result of std::accumulate is cast to `size_t`. `1` should be `size_t` to avoid possibility of bad behaviour for really big vectors.